src/services/Events.ts

```diff
@@ -5,70 +5,121 @@ import { StreamingServiceId } from '../streaming-services/streaming-services';
 import { StorageValuesOptions, StorageValuesSyncOptions } from './BrowserStorage';
 import { Errors } from './Errors';
 import { RequestException } from './Requests';
-
-export enum Events {
```

Comment:
I don't see the benefit of using an enum here, is there one? Using a conditional type makes it much easier to do generic typings.